dashboard/app/models/levels/level.rb

```diff
@@ -712,12 +712,35 @@ def show_help_and_tips_in_level_editor?
     !(is_a?(NetSim) || is_a?(GamelabJr) || is_a?(Dancelab) || is_a?(BubbleChoice))
   end
 
+  # Returns the localized teacher_markdown for the level based on the current
+  # locale.
+  #
+  # Will either return the localization from the `teacher_markdown` translations
+  # or from the `teacher_markdown` section in the `dsls` translations, with a
+  # preference for the former.
   def localized_teacher_markdown
     if should_localize?
+      # This will look up a dedicated teacher_markdown entry or the one for the
+      # dsl property, given preference to the former that might be found in a
+      # `teacher_markdown.{locale}.json` file. Otherwise, it will default to any
+      # found in a `dsls.{locale}.yml` file, which is the general place where
+      # all localized properties typically are found.
+      #
+      # Surveys use properties via dsls to localize teacher_markdown entries to
+      # describe the purpose of the survey, hence the default fallback to look
+      # for an entry off of the `dsls` grouping. Scripts/Units themselves do not
+      # store such markdown the same way. Thus they could be found in two
+      # different places.
+
+      # It is possible for this method to be called on a Level that does not
+      # have a DSL inteface and, due to this, no `localized_property` method.
+      fallback = try(:localized_property, 'teacher_markdown')
```

Comment:
I am not certain about an answer to this: is overriding this method with this logic in the `DSLDefined` model an option? Maybe that doesn't make sense because there are other Level subclasses that need this

Reply:
That's certainly possible. It feels a little weird to override it. The only places I see that call this method that don't really assume DSLDefined is in setting up appOptions and it is looking at general `Level`s and then being very accepting of a nil value being returned. Perhaps, if there should be a `try` *at all*, the `try` should be in that place and the method bumped only into places where there would be a `localized_property` method.

Another option is to special case `teacher_markdown` in the `sync-in` such that DSLDefined levels pull that into the `i18n_strings` outside of the `dsls` section in order to coerce all `teacher_markdown` to go into its own place. That starts to feel like a reasonable solution even though it makes "where localized strings are?" more maddeningly complex.

Reply:
Got it. I think this solution is good 👍